src/editors/object.js

```diff
@@ -1039,7 +1039,7 @@ export var ObjectEditor = AbstractEditor.extend({
     if (this.jsoneditor.options.remove_empty_properties || this.options.remove_empty_properties) {
       for (var i in result) {
         if (result.hasOwnProperty(i)) {
-          if ((typeof result[i] === 'undefined' || result[i] === '' || result[i] === Object(result[i])) && Object.keys(result[i]).length === 0 && result[i].constructor === Object) {
+          if (typeof result[i] === 'undefined' || result[i] === '' || (result[i] === Object(result[i]) && Object.keys(result[i]).length === 0 && result[i].constructor === Object)) {
```

Comment:
@pmk65 @sirockin @germanbisurgi @amk0 What do you think about having methods for complicated checks like this? eg.

```JS
function hasResult() {
  return (
    typeof result[i] === 'undefined' || result[i] === '' || 
    (
      result[i] === Object(result[i]) && 
      Object.keys(result[i]).length === 0 && 
      result[i].constructor === Object
    )
}
```

Also with a comment, it's pretty hard to grasp what's going on.

Reply:
But isn't the function only usable in this exact situation? Or is the same check done in other parts of the code?

Reply:
> But isn't the function only usable in this exact situation? 

Very likely :)

> Or is the same check done in other parts of the code?

I am mostly concerned about readability.

Reply:
We could format it like this (I can't remember if the &&/ || have to be placed at start or end to keep standardjs happy)
````javascript
if (typeof result[i] === 'undefined' 
  || result[i] === '' 
  || (result[i] === Object(result[i]) 
    && Object.keys(result[i]).length === 0 
    && result[i].constructor === Object)) {
````

Reply:
> Very likely :)
I looked through code for `Object.keys` uses and even when there are similar structures, I couldn't identify a common denominator and  there is no another exact case like this.

Reply:
> I am mostly concerned about readability.

Me too...but this code is in already small function (getValue), may be that just one comment be enough. In fact, lack of readability was the source of the error.

Reply:
just thinking loud...

```JS
hasResult = (typeof result[i] === 'undefined' 
  || result[i] === '' 
  || (result[i] === Object(result[i]) 
    && Object.keys(result[i]).length === 0 
    && result[i].constructor === Object))

if (hasResult)
```

Reply:
I have changed the layout, but creating a function doesn't make it more readable IMO.

Reply:
"sidequestion": What does this actually do? :smiley_cat:

Reply:
It's related to `remove_empty_properties` option, See #589

Reply:
i really thing that check should go into a function because the function name will tell more about it, it will be encapsulated (secure) and with comments will be lot nicer :)

Reply:
@germanbisurgi I don't know enough about the functionality of this check to write a comments. (I just followed @pmadril advice in #589 and tested that it worked.)
So if you want, you can make modifications to the PR.

Reply:
I also don't know enough about this particular code, just stumbled with the error and debug it up to find what's going on. Lack of readability was the problem and I think that the new layout from @pmk65 helps a lot to solve it. Could it be merged?